lib/flakes.nix

```diff
@@ -19,4 +19,24 @@ rec {
     in
     self;
 
+  /* Generates an attrset for the given list of system strings and instantiates
+    nixpkgs from a given flake. The system architecture string for each set is
+    available via the 'system' argument. The instantiated nixpkgs set is available
+    via the 'pkgs' argument.
+
+    Type: forSystems :: [String] -> Flake -> ( String -> AttrSet -> t ) -> AttrSet
+
+    Example:
+      nixpkgs.lib.forSystems
```

Comment:
This example can be simplified to `builtins.mapAttrs (system: pkgs: { x = pkgs.hello; }) (builtins.getFlake "github:nixos/nixpkgs").legacyPackages`

Reply:
For me, this is about readability. The simplification you've provided whilst equivalent, is less readable, in my view. Since it is not a function with a name, or arguments with names that explain things in code.

Reply:
Actually, I take that back. Your simplification is genius.